browser/typescript/v2/react-app/src/ampli/index.ts

```diff
@@ -0,0 +1,747 @@
+
+  constructor(
+    public event_properties: EventObjectTypesProperties,
+  ) {}
+}
+
+export class EventWithAllProperties implements BaseEvent {
+  event_type = 'Event With All Properties';
+  event_properties: EventWithAllPropertiesProperties & {
+    'requiredConst': "some-const-value";
+  };
+
+  constructor(
+    event_properties: EventWithAllPropertiesProperties,
+  ) {
+    this.event_properties = {
+      ...event_properties,
+      'requiredConst': "some-const-value",
+    };
+  }
+}
+
+export class EventWithArrayTypes implements BaseEvent {
+  event_type = 'Event With Array Types';
+
+  constructor(
+    public event_properties: EventWithArrayTypesProperties,
+  ) {}
+}
+
+export class EventWithConstTypes implements BaseEvent {
+  event_type = 'Event With Const Types';
+  event_properties = {
+    'Boolean Const': true,
+    'Integer Const': 10,
+    'Number Const': 2.2,
+    'String Const': "String-Constant",
+    'String Const WIth Quotes': "\"String \"Const With\" Quotes\"",
+    'String Int Const': 0,
+  };
+}
+
+export class EventWithDifferentCasingTypes implements BaseEvent {
+  event_type = 'event withDifferent_CasingTypes';
+
+  constructor(
+    public event_properties: EventWithDifferentCasingTypesProperties,
+  ) {}
+}
+
+export class EventWithEnumTypes implements BaseEvent {
+  event_type = 'Event With Enum Types';
+
+  constructor(
+    public event_properties: EventWithEnumTypesProperties,
+  ) {}
+}
+
+export class EventWithOptionalArrayTypes implements BaseEvent {
+  event_type = 'Event With Optional Array Types';
+
+  constructor(
+    public event_properties?: EventWithOptionalArrayTypesProperties,
+  ) {}
+}
+
+export class EventWithOptionalProperties implements BaseEvent {
+  event_type = 'Event With Optional Properties';
+
+  constructor(
+    public event_properties?: EventWithOptionalPropertiesProperties,
+  ) {}
+}
+
+// prettier-ignore
+export class Ampli {
+  private disabled: boolean = false;
+  private amplitude?: AmplitudeClient;
+
+  get client(): AmplitudeClient {
+    this.isInitializedAndEnabled();
+    return this.amplitude!;
+  }
+
+  private isInitializedAndEnabled(): boolean {
+    if (!this.amplitude) {
+      console.error('ERROR: Ampli is not yet initialized. Have you called ampli.load() on app start?');
+      return false;
+    }
+    return !this.disabled;
+  }
+
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   *
+   * @param options Configuration options to initialize the Ampli SDK with.
+   */
+  load(options?: LoadOptions) {
+    this.disabled = options?.disabled ?? false;
+
+    if (this.amplitude) {
+      console.warn('WARNING: Ampli is already intialized. Ampli.load() should be called once at application startup.');
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
```

Comment:
Need to check if the instance have the configuration or not. If not pass the deafultConfiguration.
Please reference the discussion in python ampli example.
https://github.com/amplitude/ampli-examples/pull/120#discussion_r909273631

Reply:
Add it into https://amplitude.atlassian.net/browse/AMP-55978 as a follow up ticket.